File: test-suite/handwritten-src/ts/PrimitivesTest.ts
import {TestCase, allTests, assertEq} from "./testutils"
import * as test from "../../generated-src/ts/test";

class PrimitivesTest extends TestCase {
    m: test.Test_statics;
    constructor(module: test.Test_module_statics) {
        super(module);
        this.m = module.testsuite;
    }
    testPrimitives() {
        var p = {
            b: true,                         // bool
            eight: 123,                      // i8
            sixteen: 20000,                  // i16
            thirtytwo: 1000000000,           // i32
            sixtyfour: BigInt("1234567890123456789"), // i64
            fthirtytwo: 0,                   // f32
            fsixtyfour: 1.23,                // f64

            oB: true,                         // optional<bool>
            oEight: 123,                      // optional<i8>
            oSixteen: 20000,                  // optional<i16>
            oThirtytwo: 1000000000,           // optional<i32>
            oSixtyfour: BigInt("1234567890123456789"), // optional<i64>
            oFthirtytwo: 0,                   // optional<f32>
            oFsixtyfour: 1.23                 // optional<f64>
        };
        // f32 roundtrip test is not applicable to js because js has no native
        // 32-bit floating point number.
        assertEq(p, this.m.TestHelpers.assortedPrimitivesId(p));
    }
}
allTests.push(PrimitivesTest);

File: test-suite/handwritten-src/swift/PrimitivesTest.swift
import XCTest
@testable import TestSuite
@testable import TestSuiteBridge

final class PrimitivesTest: XCTestCase {
    func testPrimitives() throws {
        let p = AssortedPrimitives(b:true, eight:123, sixteen:20000, thirtytwo:1000000000, sixtyfour:1234567890123456789, fthirtytwo:1.23, fsixtyfour:1.23,
                                   oB:true, oEight:123, oSixteen:20000, oThirtytwo:1000000000, oSixtyfour:1234567890123456789, oFthirtytwo:1.23, oFsixtyfour:1.23)
        XCTAssertEqual(p, try TestHelpers_statics.assortedPrimitivesId(p))
    }
}
